dashboard/test/ui/step_definitions/publicKeyCryptography_steps.rb

```diff
@@ -10,3 +10,88 @@
     And I rotate to landscape
   STEPS
 end
+
+When /^I set the modulo clock speed to ([1-9])$/ do |speed|
+  set_react_select(@browser.find_element(:css, '.speed-dropdown'), speed)
+end
+
+When /^I set the clock size to (\d+)$/ do |clock_size|
+  clock_size_input = @browser.find_element(:css, '.modulus-textbox')
+  clock_size_input.click
+  10.times { clock_size_input.send_key(:backspace) }
+  clock_size_input.send_keys(clock_size.to_s)
+end
+
+When /^I set the dividend to (\d+)$/ do |dividend|
+  clock_size_input = @browser.find_element(:css, '.dividend-textbox')
+  clock_size_input.click
+  10.times { clock_size_input.send_key(:backspace) }
+  clock_size_input.send_keys(dividend.to_s)
+end
+
+When /^I run the modulo clock$/ do
+  @browser.find_element(:css, '.go-button').click
+end
+
+When /^I (?:hide|show) Eve's panel$/ do
+  @browser.find_element(:xpath, "//div[text()='Eve']").click
+  sleep 0.5
+end
+
+When /^I click the start over button$/ do
+  @browser.find_element(:xpath, "//button[text()='Start Over']").click
+end
+
+When /^I click the OK button in the dialog$/ do
+  @browser.find_element(:xpath, "//button[text()='OK']").click
+end
+
+When /^I set the public modulus to (\d+)$/ do |modulus|
+  set_react_select(@browser.find_element(:css, '.panel-eve .public-modulus-dropdown'), modulus)
+end
+
+When /^Alice sets her private key to (\d+)$/ do |private_key|
+  set_react_select(@browser.find_element(:css, '.panel-alice .private-key-dropdown'), private_key)
+end
+
+When /^Bob sets his secret number to (\d+)$/ do |value|
+  set_react_select(@browser.find_element(:css, '.panel-bob .secret-number-dropdown'), value)
+end
+
+When /^Eve sets Bob's secret number to (\d+)$/ do |value|
+  set_react_select(@browser.find_element(:css, '.panel-eve .secret-number-dropdown'), value)
+  sleep 6
+end
+
+When /^I calculate Bob's public number$/ do
+  @browser.find_element(:css, '.panel-bob button').click
+  sleep 6
+end
+
+When /^Alice decodes Bob's secret number$/ do
+  @browser.find_element(:css, '.panel-alice button').click
+  sleep 6
```

Comment:
Is it really necessary to sleep for six seconds in these tests?

Reply:
These are waiting on an animation to complete.  If you'd like to speed these tests up, we'll need to expose a JavaScript hook in the PKC widget that allows us to accelerate this animation or skip it entirely.